packages/chain-adapters/src/utxo/UTXOBaseAdapter.ts

```diff
@@ -0,0 +1,161 @@
+import {
+  accountTypeToScriptType,
+  bnOrZero,
+  convertXpubVersion,
+  toRootDerivationPath
+} from '../utils'
+
+export type UTXOChainTypes = ChainTypes.Bitcoin // to be extended in the future to include other UTXOs
+
+/**
+ * Currently, we don't have a generic interact for UTXO providers, but will in the future.
+ * Leaving this as-is for now, but we will need to test in the future when we have additional
+ * UTXO chains implemented in unchained.
+ */
+export interface ChainAdapterArgs {
+  providers: {
+    http: bitcoin.api.V1Api //?
+    ws: bitcoin.ws.Client
+  }
+  coinName: string
+  chainId?: CAIP2
+}
+
+/**
+ * Base chain adapter for all UTXO chains. When extending please add your ChainType to the
+ * UTXOChainTypes. For example:
+ *
+ * `export type UTXOChainTypes = ChainTypes.Bitcoin | ChainTypes.Litecoin`
+ */
+export abstract class UTXOBaseAdapter<T extends UTXOChainTypes> implements IChainAdapter<T> {
+  protected chainId: CAIP2
+  protected coinName: string
+  protected readonly providers: {
+    http: bitcoin.api.V1Api
+    ws: bitcoin.ws.Client
+  }
+
+  protected constructor(args: ChainAdapterArgs) {
+    this.providers = args.providers
+  }
+
+  /* Abstract Methods */
+
+  abstract subscribeTxs(
+    input: chainAdapters.SubscribeTxsInput,
+    onMessage: (msg: chainAdapters.SubscribeTxsMessage<T>) => void,
+    onError?: (err: chainAdapters.SubscribeError) => void
+  ): Promise<void>
+  abstract unsubscribeTxs(input?: chainAdapters.SubscribeTxsInput): void
+  abstract closeTxs(): void
+  abstract getType(): T
+
+  abstract getTxHistory(
+    input: chainAdapters.TxHistoryInput
+  ): Promise<chainAdapters.TxHistoryResponse<T>>
+
+  abstract buildBIP44Params(params: Partial<BIP44Params>): BIP44Params
+
+  abstract buildSendTransaction(
+    tx: chainAdapters.BuildSendTxInput<T>
+  ): Promise<{ txToSign: chainAdapters.ChainTxType<T> }>
+
+  abstract getAddress(input: chainAdapters.GetAddressInput): Promise<string>
+
+  abstract getFeeData(
+    input: Partial<chainAdapters.GetFeeDataInput<T>>
+  ): Promise<chainAdapters.FeeDataEstimate<T>>
+
+  abstract signTransaction(
+    signTxInput: chainAdapters.SignTxInput<chainAdapters.ChainTxType<T>>
+  ): Promise<string>
+
+  /* public methods */
+
+  getCaip2(): CAIP2 {
+    return this.chainId
+  }
+
+  getChainId(): CAIP2 {
+    return this.chainId
+  }
+
+  async getAccount(pubkey: string): Promise<chainAdapters.Account<T>> {
+    if (!pubkey) {
+      return ErrorHandler('UTXOBaseAdapter: pubkey parameter is not defined')
+    }
+
+    try {
+      const caip = await this.getCaip2()
+      const { chain, network } = caip2.fromCAIP2(caip)
+      const { data } = await this.providers.http.getAccount({ pubkey: pubkey })
+
+      const balance = bnOrZero(data.balance).plus(bnOrZero(data.unconfirmedBalance))
+
+      return {
+        balance: balance.toString(),
+        chain: this.getType(),
+        caip2: caip,
+        caip19: caip19.toCAIP19({
```

Comment:
Another TODO is we should probably set `assetId` similarly to keeping track of `chainId` that we would be able to use for the native asset caip19 as determined by the implementing class. This would remove the Bitcoin reference in the base adapter

Reply:
I am not sure I am following.  The bitcoin reference in the base adapter is just for the provider. Can you clarify what you are asking to change here?

Reply:
I probably should have made the comment on a better line. See here: https://github.com/shapeshift/lib/pull/411/files/784d591ee848ebe7444da8bf528a595c0727bde7#diff-91f7561fb5e5062d3ca7bc6c42e96c285c398a4518d8485bdcb9f672cb4148d7R111

We are using `AssetReference.Bitcoin` in the `UTXOBaseAdapter` to create the caip19 string, whereas I am suggesting that if we add an `assetId` (bitcoin caip19) class variable that is set from the bitcoin `ChainAdapter` class, we can use `this.assetId` which will be of the proper type as determined by the implementing class.

Reply:
Got it, make sense. please take a look at the latest commit

Reply:
That looks great!

TBD if we want to keep `getCaip19()`. We have both for caip2 since the `getCaip2()` was added to the base `ChainAdapter` interface, but if we are trying to move naming to `chainId` and `assetId` we may just want the `getAssetId()` function. I haven't been following super closely with that naming discussion, so may be good for @0xdef1cafe or whoever else to weigh in.

You have my stamp for everything as is and we can iterate as needed from here.

Reply:
Thanks a ton for all the help and reviews. @0xdef1cafe - lmk your preference here and we can merge this so I don't have to fix yet another conflict ;)